grpc-server/grpc_helpers_test.go

```diff
@@ -0,0 +1,232 @@
+package grpcserver
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/packethost/cacher/protos/cacher"
+	"github.com/packethost/pkg/grpc"
+	"github.com/packethost/pkg/log"
+	assert "github.com/stretchr/testify/require"
+	"github.com/tinkerbell/hegel/grpc/protos/hegel"
+	"github.com/tinkerbell/hegel/hardware"
+	ggrpc "google.golang.org/grpc"
+	"google.golang.org/grpc/status"
+	"google.golang.org/grpc/test/bufconn"
+)
+
+const bufSize = 1024 * 1024
+
+type zapT struct {
+	*testing.T
+	mu   sync.RWMutex
+	done bool
+}
+
+// Logs the given message without failing the test.
+func (zt *zapT) Logf(format string, args ...interface{}) {
+	zt.mu.RLock()
+	if !zt.done {
+		zt.T.Logf(format, args...)
+	}
+	zt.mu.RUnlock()
+}
+
+// Logs the given message and marks the test as failed.
+func (zt *zapT) Errorf(format string, args ...interface{}) {
+	zt.mu.RLock()
+	if !zt.done {
+		zt.T.Logf(format, args...)
+	}
+	zt.mu.RUnlock()
+}
+
+// serveAndDial starts 2 grpc services.
```

Comment:
should be `startServersAndConnectClient`

Reply:
And the rest of the comment updated too. 😄

Reply:
Done